docs/reference/files/charmcraft-yaml-file.rst

```diff
@@ -424,6 +432,46 @@ secret URI.
         :start-at: config:
         :end-before: containers:
 
+.. admonition:: Best Practice
+    :class: hint
+
+    Charms should keep the user experience of the operator as simple and obvious
+    as possible. Aim to make the application deployable without providing any
+    further configuration options: ``juju deploy foo`` is preferable to
+    ``juju deploy foo --config required-option=value``.
+
```

Comment:
You might streamline this as: To make the user's experience as smooth as possible, aim to...